jetstream/cli.py

```diff
@@ -239,52 +277,65 @@ def convert(self, value, param, ctx):
 )
 @experiment_slug_option
 @secret_config_file_option
-def run(project_id, dataset_id, date, experiment_slug, config_file):
-    """
-    Runs analysis on active experiments for the provided date.
-
-    This command is invoked by Airflow. All errors are written to the console and
-    BigQuery. Runs will always return a success code, even if exceptions were
-    thrown during some experiment analyses. This ensures that the Airflow task will
-    not retry the task and run all of the analyses again.
-    """
-    AnalysisExecutor(
+@argo_option
+@zone_option
+@cluster_id_option
+@monitor_status_option
+def run(
+    project_id,
+    dataset_id,
+    date,
+    experiment_slug,
+    config_file,
+    argo,
+    zone,
+    cluster_id,
+    monitor_status,
+):
+    """Runs analysis for the provided date."""
+    strategy = SerialExecutorStrategy(project_id, dataset_id)
+    if argo:
+        strategy = ArgoExecutorStrategy(project_id, zone, cluster_id, monitor_status)
+
+    success = AnalysisExecutor(
         project_id=project_id,
         dataset_id=dataset_id,
         date=date,
         experiment_slugs=[experiment_slug] if experiment_slug else All,
         configuration_map={experiment_slug: config_file} if experiment_slug and config_file else {},
-    ).execute()
+    ).execute(strategy=strategy)
 
-    # todo: uncomment for Argo
-    # sys.exit(0 if success else 1)
+    sys.exit(0 if success else 1)
```

Comment:
One thing to consider here, Argo will also retry the pod if there is any exception being thrown (like a `UnenrollmentPeriodMissingException` and others)

Reply:
We may as well avoid that, right?

Reply:
(I'm fine doing this later)